Fix a hang observed on a Tiger/PPC system when attempting to end a debugging
session. pthread_detach is essentially an asynchronous pthread_join, so this
shouldn't have any side effects, but apply at your own discretion.

--- gdb/macosx/macosx-nat-threads.c.orig
+++ gdb/macosx/macosx-nat-threads.c
@@ -46,7 +46,7 @@
 
   ret = pthread_cancel (pthread);
 
-  ret = pthread_join (pthread, NULL);
+  ret = pthread_detach (pthread);
   if (ret != 0)
     {
       warning ("Unable to join to canceled thread: %s (%d)", strerror (errno),
